common/persistence/cassandra/cassandraMetadataPersistenceV2.go

```diff
@@ -90,10 +90,8 @@ func (m *nosqlDomainManager) CreateDomain(
 	err = m.db.InsertDomain(ctx, row)
 
 	if err != nil {
-		if m.db.IsConditionFailedError(err) {
```

Comment:
did we reproduce this line on prod?

Reply:
We don't have concurrent domain registration on prod, so won't be able to repro there. I can easily repro this on my local though, for example starting canary with two non-exists domains in the config.